src/Features/Core/Portable/EditAndContinue/CommittedSolution.cs

```diff
@@ -427,11 +436,21 @@ await TryGetMatchingSourceTextAsync(log, sourceText, sourceFilePath, currentDocu
         }
     }
 
-    public void CommitSolution(Solution solution)
+    public void CommitChanges(Solution solution, ImmutableArray<ProjectId> projectsToStale, ImmutableArray<ProjectId> projectsToUnstale)
```

Comment:
do you want any asserts on these input arrays?  for example, that tjhey shouldn't overlap?  or, if they can overlap, doc as such to make semantics clear.